chia/consensus/blockchain.py

```diff
@@ -735,13 +735,19 @@ def clean_block_record(self, height: int) -> None:
         Args:
             height: Minimum height that we need to keep in the cache
         """
+        if self._peak_height is not None and height > self._peak_height - self.constants.BLOCKS_CACHE_SIZE:
+            height = self._peak_height - self.constants.BLOCKS_CACHE_SIZE
         if height < 0:
             return None
+        ses_heights = self.get_ses_heights()[-4:]
```

Comment:
your intention here is just to leave the records last blocks that correspond with the last four sub epoch summaries ?

that seems a bit weird since the cache under the peak is larger then 4 sub epochs and these should always be in the cache anyway 


is this just for tests ?

Reply:
fundamentally, the intention is to allow `create_prev_sub_epoch_segments()` to not fail with the `KeyError` that I mention in the description. These blocks are required to be in the cache for that, so, yes.

> that seems a bit weird since the cache under the peak is larger then 4 sub epochs and these should always be in the cache anyway

Are you sure? As you can see, I also make sure we keep `self.constants.BLOCKS_CACHE_SIZE` block records below the current peak. That was not enough to avoid the error.

> is this just for tests ?

It's possible that there's a quirk with the test constants that makes this problem only appear in tests.

Reply:
i dont like that we add production code specifically so tests would work :\
if you really insist i guess we can but i feel like a better approach is fixing the test quirk

Reply:
I added a commit to increase test block cache size in simulations, and remove the special case for blocks with sub-epoch-summaries